src/components/Api.js

```diff
@@ -0,0 +1,61 @@
+export default class Api {
+  constructor(options) {
+    this._baseUrl = options.baseUrl;
+    this._headers = options.headers;
+  }
+
+  _getRequest(endPoint, method = "GET") {
+    return fetch(this._baseUrl + endPoint, {
+      method: method,
+      headers: this._headers,
+    }).then(this._checkSuccessResponce);
+  }
+
+  _checkSuccessResponce(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+  _requestWithBody(endPoint, body, method = "POST") {
+    return fetch(this._baseUrl + endPoint, {
+      method: method,
+      headers: this._headers,
+      body: JSON.stringify(body),
+    })
+      .then(this._checkSuccessResponce)
+      .catch((err) => {
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request and this end is in `index.js`  after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.